ext/openssl/ossl_pkey.c

```diff
@@ -532,6 +532,52 @@ ossl_pkey_initialize(VALUE self)
     return self;
 }
 
+/*
+ *  call-seq:
+ *      OpenSSL::PKey.private_new(algo, string) -> PKey
+ *
+ * See the OpenSSL documentation for EVP_PKEY_new_raw_private_key()
+ */
+
+static VALUE
+ossl_pkey_initialize_private(VALUE self, VALUE type, VALUE key)
+{
+    EVP_PKEY *pkey;
+
+    int nid = OBJ_sn2nid(StringValueCStr(type));
+    if(!nid) ossl_raise(ePKeyError, "unknown OID `%"PRIsVALUE"'", type);
```

Comment:
Please use EVP_PKEY_asn1_find_str() instead of OBJ_sn2nid(). 

pkey_generate() is currently doing this. I think it should be extracted as a function.